smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptWriter.java

```diff
@@ -114,19 +142,54 @@ public TypeScriptWriter addIgnoredDefaultImport(String name, String from, String
      */
     @Deprecated
     public TypeScriptWriter addImport(String name, String as, String from) {
+        boolean isNodePackage = from.startsWith("node:");
```

Comment:
should a class wrap the String input `from` to provide answers like `boolean isNamespaced`, `boolean isNodeJSNativePackage`, `String getPackageName` etc.?

Reply:
That probably should go in `Dependency` in the long run.